src/app/submit-text-message-template/[sessionId]/page.tsx

```diff
@@ -0,0 +1,28 @@
+'use server';
+
+import { redirect, RedirectType } from 'next/navigation';
+import { SubmitTemplate } from '@forms/SubmitTemplate/SubmitTemplate';
+import { PageProps, Session, TemplateType } from '@utils/types';
+import { getSession } from '@utils/form-actions';
+
+const isValid = (session?: Session) =>
+  session?.templateType === TemplateType.SMS &&
+  session?.smsTemplateMessage !== undefined &&
+  session?.smsTemplateName !== undefined;
+
+const SubmitSmsTemplatePage = async ({ params: { sessionId } }: PageProps) => {
+  const session = await getSession(sessionId);
+
+  if (!session || !isValid(session)) {
+    return redirect('/invalid-session', RedirectType.replace);
+  }
+
+  return (
+    <SubmitTemplate
+      templateName={session.smsTemplateName!}
```

Comment:
i'm personally not a fan of '!' typecasts (also seen in lots of other places) but probably doesn't matter if it's not easy to sort

Reply:
I completely agree. I don't like it. 

I think we need a bit of a change to the session model.